libnetwork/osl/namespace_linux.go

```diff
@@ -338,6 +313,39 @@ func createNamespaceFile(path string) (err error) {
 	return err
 }
 
+// The networkNamespace type is the linux implementation of the Sandbox
+// interface. It represents a linux network namespace, and moves an interface
+// into it when called on method AddInterface or sets the gateway etc.
+type networkNamespace struct {
+	path         string
+	iFaces       []*nwIface
+	gw           net.IP
+	gwv6         net.IP
+	staticRoutes []*types.StaticRoute
+	neighbors    []*neigh
+	nextIfIndex  map[string]int
+	isDefault    bool
+	nlHandle     *netlink.Handle
+	loV6Enabled  bool
+	sync.Mutex
+}
+
+func (n *networkNamespace) Interfaces() []Interface {
+	ifaces := make([]Interface, len(n.iFaces))
+	for i, iface := range n.iFaces {
+		ifaces[i] = iface
+	}
+	return ifaces
+}
+
+func (n *networkNamespace) InterfaceOptions() IfaceOptionSetter {
```

Comment:
Oh libnetwork... This makes no sense!

https://github.com/moby/moby/blob/2cccb1f02c83aaacef3c15fa43f3b64d57f315f8/libnetwork/osl/options_linux.go#L13-L73

Reply:
It doesn't. There's also fun things where the interface defines a function that defines an interface.

And it turns out that interface is just an easy way _NOT_ to have to implement it, because it just returns `nil` on Windows, which returns a `nil` interface... which happens to not make things fail, because the code is never called on Windows 🙈 

- More of that to come once I fixed the remaining issues in https://github.com/moby/moby/pull/46149